### ISSUE TITLE
Error while following custom component tutorial.

### ISSUE THREAD

**Issue by arcontechnologies** [gradio-app/gradio#8984]
### Describe the bug

following custom component tutorial, I started by creating the project :
`gradio cc create MyTextBox --template SimpleTextbox`

then I tried run the next command : `gradio cc dev`  where an error occurred :  

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

demo code

### Screenshot

![image](https://github.com/user-attachments/assets/dff14fe4-2d04-440c-9810-624ad64d8678)

![image](https://github.com/user-attachments/assets/3e3aa59f-8a3d-436c-ad63-778653125263)

### System Info

```shell
Windows 11
python 3.11
Nodejs : v18.14.2
npm : 9.5.1
```

### Severity

Blocking usage of gradio

**Comment by abidlabs** [comment]
Hi @arcontechnologies are you using the latest version of `gradio`?

**Comment by arcontechnologies** [comment]
Hi @abidlabs Yes I'm using the last one : 4.40.0
I tried 4.39.0 and also 4.35.0 to see if any regression but I'm ending with same error.

**Comment by abidlabs** [comment]
Is this the case if you use a different component as a template as well?

**Comment by arcontechnologies** [comment]
Yes I tried with **File** (following tutorial recommendation) and got the same outcome.

**Comment by freddyaboulton** [comment]
Hi @arcontechnologies - What version do you have for `@gradio/preview` in `frontend/package.json`?

**Comment by arcontechnologies** [comment]
here it is : "@gradio/preview": "0.10.2"

**Comment by arcontechnologies** [comment]
@freddyaboulton Hi, any news regarding this ? Let me know if I have to handle further tests or check something else.

**Comment by PietFourie** [comment]
I do not know if I had the same problem but .....
I was also following the basic tutorial and created a custom component from a fileExplorer component. It did not work out of the box running your steps above, even if I changed nothing .......
The problem is I am running linuxmint.
What you will find is that there is a "requirements.txt" file in the demo folder. This means the component must be installed in python first. Usually you would do it as follows.... python install  -e mycomponent, which means you can edit the python code and then it is automatically updated in your python installed folder. But this does not work here.
What you have to do, (in my case) run "gradio cc install" first, which installs your component in the python library. Only then can the app.py find it and run it using "gradio cc dev". The developers must update the "create" scrypt or include this in the documentation ....
This solved my problem. Hope it does yours.
BTW I first created a conda environment to make sure everything is OK....

**Comment by PietFourie** [comment]
Got another error on linux during build. I got "permission" problems
I had to install the following ......
sudo pip install trove-classifiers tomli pluggy pathspec packaging hatchling hatch-requirements-txt hatch-fancy-pypi-readme

**Comment by freddyaboulton** [comment]
Can you please try with the latest version of gradio? I think this issue has been fixed!
